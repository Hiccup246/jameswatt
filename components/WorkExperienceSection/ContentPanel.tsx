import React from "react"

// import './content-panel.css'

export default function ContentPanel({ job, activated }: { job:{ role:string, company:string, dateRange:string, description:string, bullets:[] }, activated:boolean }) {
  return (
    <div className={activated ? "tab-panel activated" : "tab-panel"}>
      <h3>
        <span className="company-role">{job.role}</span>
        <span className="company-name"> @ {job.company}</span>
      </h3>
      <p className="date-range">{job.dateRange}</p>
      <div className="company-description">{job.description}</div>
      <ul className="experience-points">
        {job.bullets.map((element, index) => {
          return <li key={index}>{element}</li>
        })}
      </ul>
    </div>
  )
}