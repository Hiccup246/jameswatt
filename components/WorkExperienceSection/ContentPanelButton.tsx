import React from "react"

export default function ContentPanelButton({ companyName, activated, clickHandler }: { companyName:string, activated: boolean, clickHandler:Function }) {
  const activatedClasses = activated ? "text-black activated" : ""

  return (
    <button
      aria-label={companyName}
      className={`
        tab-item
        cursor-pointer relative flex w-fit
        h-[var(--tab-height)]
        leading-[var(--tab-height)] px-5 bg-transparent
        whitespace-nowrap z-20 text-black
        mb-[var(--tab-margin-top)] focus-visible:border-none
        max-small:mb-0 max-small:max-w-[var(--tab-width)]
        max-small:grow-0 max-small:shrink-0 max-small:basis-[var(--tab-width)]
        max-small:justify-center
        ${activatedClasses}
      `}
      onClick={(element) => clickHandler(element.currentTarget.offsetWidth)}
    >
      {companyName}
    </button>
  )
}