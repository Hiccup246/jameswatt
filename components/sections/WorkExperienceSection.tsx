import { PROGRAMMING_EXPERIENCES } from "../../constants/WorkExperiences";
import SectionLayout from "../layouts/SectionLayout";
import TabsManager from "../TabsManager";

export default function WorkExperienceSection() {
  return (
    <SectionLayout>
      <h1 className="my-16 text-center text-2xl font-bold sm:text-3xl">
        My Experience
      </h1>

      <TabsManager>
        {PROGRAMMING_EXPERIENCES.map((job: Job) => {
          return (
            <div key={`${job.company}-${Math.random().toFixed(3)}`}>
              <h2 className="m-0">
                <span className="text-xl font-bold">{job.role}</span>
                <span className="text-xl font-bold"> @ {job.company}</span>
              </h2>
              <p className="mt-1 mb-4">{job.dateRange}</p>
              <div className="pb-4">{job.description}</div>
              <ul className="m-0 list-none p-0">
                {job.jobHighlights.map((highlight: string) => {
                  return (
                    <li
                      className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['â†’']"
                      key={"job-highlight" + Math.random()}
                      dangerouslySetInnerHTML={{ __html: highlight }}
                    ></li>
                  );
                })}
              </ul>
            </div>
          );
        })}
      </TabsManager>
    </SectionLayout>
  );
}
