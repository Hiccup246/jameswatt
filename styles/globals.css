@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  :root {
    --slider-items: 7;
    --slider-item-sets: 2;
    --all-slider-items: calc(var(--slider-items) * var(--slider-item-sets));
    --slider-item-width: 200px;
  }

  html,
  body {
    padding: 0;
    margin: 0;
    font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont,
      "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif,
      "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol",
      "Noto Color Emoji";
    @apply bg-brown;
  }

  @property --gradient-angle {
    syntax: "<angle>";
    inherits: false;
    initial-value: 1turn;
  }
}

@layer utilities {
  .animation-pause {
    animation-play-state: paused;
  }

  .animation-play {
    animation-play-state: running;
  }
}
