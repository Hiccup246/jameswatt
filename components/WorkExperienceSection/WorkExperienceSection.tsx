import React, { useEffect, useState, useRef } from "react"

// import './programming-history.css'
import { PROGRAMMING_EXPERIENCES } from "../../constants/WorkExperiences"
import ContentPanelButton from "./ContentPanelButton"
import ContentPanel from "./ContentPanel"
import SectionLayout from "../SectionLayout"

export default function ExperienceSection() {
  const [currentTabIndex, setCurrentTabIndex] = useState<number>(0)
  const [sliderStyle, setSliderStyle] = useState<{ width: string, transform:string }>({ width: "", transform: "" })

  const tabComponentRef = useRef<HTMLDivElement>(null)
  const mobileViewWidth = "550px"

  const setupComponent = () => {
    const panelContainer:HTMLElement = tabComponentRef.current as HTMLElement;

    const tabItem:HTMLElement = panelContainer.querySelector(
      ".tab-item.activated"
    ) as HTMLElement

    const selectedTabItem:HTMLElement = panelContainer.querySelector(
      ".selected-tab-item"
    ) as HTMLElement

    const newSliderWidth:number = tabItem.clientWidth

    selectedTabItem.style.width = `${newSliderWidth}px`
    
    panelContainer.style.height = largestPanelHeight()
  }

  const largestPanelHeight = () => {
    let largestPanelHeight = 0

    for (let i = 0; i < PROGRAMMING_EXPERIENCES.length; i++) {
      const panelContainer:HTMLElement = tabComponentRef.current as HTMLElement;
      const tabPanel:NodeListOf<HTMLElement> = panelContainer.querySelectorAll(
        ".tab-panel"
      ) as NodeListOf<HTMLElement>

      // console.log(panelContainer.clientHeight)
      const panelHeight = tabPanel[i].clientHeight
      console.log(panelHeight)


      if (panelHeight > largestPanelHeight) {
        largestPanelHeight = panelHeight
      }
    }

    return `${largestPanelHeight}px`
  }

  const clickOnTab = (newTabIndex:number, buttonWidth:number) => {
    updateSelectedTabItem(newTabIndex, `${buttonWidth}px`)
    setCurrentTabIndex(newTabIndex)
  }

  const updateSelectedTabItem = (newTabIndex:number, newSliderWidth:string) => {
    const isMobile = window.matchMedia(
      `(max-width: ${mobileViewWidth})`
    ).matches
    let sliderTransformString

    if (isMobile) {
      // If the device is a mobile the tabs are horizontal
      const paddingX = `calc(${newTabIndex}*var(--tab-width))`
      const transformXString = `translateX(${paddingX})`
      sliderTransformString = transformXString
    } else {
      // If the device is a tablet or dekstop then the tabs are vertical
      const paddingY = `calc(${newTabIndex}* var(--tab-margin-top))`
      const heightGap = `calc(${newTabIndex}* var(--tab-height))`
      const transformYString = `translateY(calc(${paddingY} + ${heightGap}))`
      sliderTransformString = transformYString
    }

    setSliderStyle({ width: newSliderWidth, transform: sliderTransformString })
  }

  useEffect(() => {
    setupComponent()
  }, [])

  return (
    <SectionLayout isBgColorPrimary={false}>
      <h1 className="self-center text-2xl sm:text-3xl font-bold mb-10">
        My Experience
      </h1>

      <div ref={tabComponentRef} className="w-full flex max-[400px]:block">
        <div className="relative w-max m-0 list-none z-10 h-fit py-2.5">
           {
            PROGRAMMING_EXPERIENCES.map((job, index) => {
              return (
                <ContentPanelButton
                  key={index + job.company}
                  companyName={job.company}
                  activated={index === currentTabIndex}
                  clickHandler={(width: number) => clickOnTab(index, width)}
                />
              )
            })
           }
          <div style={sliderStyle.width.length > 0 ? sliderStyle: {}}
            className={`
              selected-tab-item
              visible absolute top-0 left-0 z-10 w-20
              h-[var(--tab-height)] my-2.5 bg-darkbrown
              transition-[transform width] duration-200 delay-100
              ease-[cubic-bezier(0.645, 0.045, 0.355, 1)]
            `}></div>
          <div className={`
            visible absolute top-0 left-0 w-0.5 bg-black
            ml-1 h-[calc(100%-var(--tab-margin-top))]
          `} />
        </div>

        <div className="relative w-full ml-5">
          {
            PROGRAMMING_EXPERIENCES.map((job: Job, index) => {
              return (
              <ContentPanel
                key={index + job.dateRange}
                job={job}
                activated={index === currentTabIndex} />
              )
            })
          }
        </div>
      </div>
    </SectionLayout>
  )
  {/* // return (
  //   <div className="programming-history">
  //     <h2>ðŸ“œ Programming History</h2>
  //     <div ref={tabComponentRef} className="tab-component">
  //       <div className="tab-menu">
  //         {PROGRAMMING_EXPERIENCES.map((job, index) => {
  //           return (
  //             <ContentPanelButton
  //               key={index + job.company}
  //               companyName={job.company}
  //               activated={index === currentTabIndex}
  //               clickHandler={(width) => clickOnTab(index, width)}
  //             />
  //           )
  //         })}

  //         <div style={sliderStyle} className="selected-tab-item"></div>
  //         <div className="menu-border" />
  //       </div>

  //       <div className="tab-content">
  //         {PROGRAMMING_EXPERIENCES.map((job, index) => {
  //           return (
  //             <ContentPanel
  //               key={index + job.dateRange}
  //               job={job}
  //               activated={index === currentTabIndex}
  //             />
  //           )
  //         })}
  //       </div>
  //     </div>
  //   </div>
  // ) */}
}