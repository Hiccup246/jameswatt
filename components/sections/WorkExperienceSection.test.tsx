import { expect, test, jest, describe } from "@jest/globals";
import WorkExperienceSection, { calcTabButtonTranslation, largestChildHeight } from "./WorkExperienceSection";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const dummyJobs: Job[] = [
  {
    role: "Head Mouser",
    company: "Dummy 1",
    dateRange: "2020/09 - 2022/10",
    description: "My first dummy job",
    jobHighlights: ["highlight 1", "highlight 2"],
  },
  {
    role: "Head Mocker",
    company: "Dummy 2",
    dateRange: "2020/01 - 2022/02",
    description: "My second dummy job",
    jobHighlights: ["highlight 1", "highlight 2"],
  },
];

test("renders the WorkExperienceSection correctly", () => {
  const tree = renderer
    .create(<WorkExperienceSection jobs={dummyJobs} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("renders the first job and all the correct tab buttons", () => {
  render(<WorkExperienceSection jobs={dummyJobs} />);

  const jobRole = screen.getByText("Head Mouser");
  const tabButtonOne = screen.getByRole("button", { name: "Dummy 1" });
  const tabButtonTwo = screen.getByRole("button", { name: "Dummy 2" });

  expect(jobRole).toBeInTheDocument();
  expect(tabButtonOne).toBeInTheDocument();
  expect(tabButtonTwo).toBeInTheDocument();
});

test("when the second tab button is clicked it renders the second job", async () => {
  Object.defineProperty(window, "matchMedia", {
    configurable: true,
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
    })),
  });

  render(<WorkExperienceSection jobs={dummyJobs} />);

  await userEvent.click(screen.getByRole("button", { name: "Dummy 2" }));

  const job = screen.getByText("Head Mocker");

  expect(job).toBeInTheDocument();
});

describe("calcTabButtonTranslation on a large device", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      configurable: true,
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
      })),
    });
  })

  test("calcTabButtonTranslation called with -1", () => {
    expect(calcTabButtonTranslation(-1)).toBe("translateY(calc((-1 * var(--tab-height)) + (-1 * var(--tab-margin-top)))")
  })
  
  test("calcTabButtonTranslation called with 0", () => {
    expect(calcTabButtonTranslation(0)).toBe("translateY(calc((0 * var(--tab-height)) + (0 * var(--tab-margin-top)))")
  })
  
  test("calcTabButtonTranslation called with 1", () => {
    expect(calcTabButtonTranslation(1)).toBe("translateY(calc((1 * var(--tab-height)) + (1 * var(--tab-margin-top)))")
  })
})

describe("calcTabButtonTranslation on a small device", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      configurable: true,
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: true,
      })),
    });
  })
  
  test("calcTabButtonTranslation called with -1", () => {
    expect(calcTabButtonTranslation(-1)).toBe("translateX(calc(-1 * var(--tab-width)))")
  })
  
  test("calcTabButtonTranslation called with 0", () => {
    expect(calcTabButtonTranslation(0)).toBe("translateX(calc(0 * var(--tab-width)))")
  })
  
  test("calcTabButtonTranslation called with 1", () => {
    expect(calcTabButtonTranslation(1)).toBe("translateX(calc(1 * var(--tab-width)))")
  })
})

test("largestChildHeight called with empty div", () => {
  const mockElement:HTMLDivElement = document.createElement("div")
  expect(largestChildHeight(mockElement)).toBe(1000)
})

test("largestChildHeight called with div that has one experience panel", () => {
  const mockElement:HTMLDivElement = document.createElement("div")
  const mockChild:HTMLDivElement = document.createElement("div")
  jest.spyOn(mockChild, 'clientHeight', 'get').mockImplementation(() => 100);
  mockElement.appendChild(mockChild)

  expect(largestChildHeight(mockElement)).toBe(100)
})

test("largestChildHeight called with div that has two experience panels", () => {
  const mockElement:HTMLDivElement = document.createElement("div")
  const mockChildOne:HTMLDivElement = document.createElement("div")
  const mockChildTwo:HTMLDivElement = document.createElement("div")

  jest.spyOn(mockChildOne, 'clientHeight', 'get').mockImplementation(() => 100);
  mockElement.appendChild(mockChildOne)

  jest.spyOn(mockChildTwo, 'clientHeight', 'get').mockImplementation(() => 200);
  mockElement.appendChild(mockChildTwo)

  expect(largestChildHeight(mockElement)).toBe(200)
})

test("largestChildHeight called with div that has two experience panels with the same heights", () => {
  const mockElement:HTMLDivElement = document.createElement("div")
  const mockChildOne:HTMLDivElement = document.createElement("div")
  const mockChildTwo:HTMLDivElement = document.createElement("div")

  jest.spyOn(mockChildOne, 'clientHeight', 'get').mockImplementation(() => 10);
  mockElement.appendChild(mockChildOne)

  jest.spyOn(mockChildTwo, 'clientHeight', 'get').mockImplementation(() => 10);
  mockElement.appendChild(mockChildOne)

  expect(largestChildHeight(mockElement)).toBe(10)
})