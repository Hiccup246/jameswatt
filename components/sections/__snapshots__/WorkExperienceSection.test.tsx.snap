// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkExperienceSection renders correctly 1`] = `
<div
  className="flex w-full flex-row justify-center pt-12 pb-20 pt-20 bg-white dark:bg-redgrey dark:text-softwhite"
>
  <section
    className="mx-auto flex w-9/12 flex-col justify-start"
  >
    <h1
      className="my-16 text-center text-2xl font-bold sm:text-3xl"
    >
      My Experience
    </h1>
    <div
      className="flex w-full max-small:block"
    >
      <div
        className="
          relative z-10 m-0 flex h-fit w-max list-none flex-col gap-6 py-2.5 max-small:flex max-small:w-full
          max-small:flex-row max-small:gap-0 max-small:overflow-y-hidden max-small:overflow-x-scroll max-small:py-0
          max-small:pb-2
        "
      >
        <button
          aria-label="AplyiD"
          className="
                bg-transparent relative z-20 flex w-fit cursor-pointer whitespace-nowrap px-5 py-2
                focus-visible:border-none max-small:mb-0 max-small:shrink-0 max-small:grow-0
                max-small:justify-center 
              "
          onClick={[Function]}
        >
          AplyiD
        </button>
        <button
          aria-label="AplyiD"
          className="
                bg-transparent relative z-20 flex w-fit cursor-pointer whitespace-nowrap px-5 py-2
                focus-visible:border-none max-small:mb-0 max-small:shrink-0 max-small:grow-0
                max-small:justify-center 
              "
          onClick={[Function]}
        >
          AplyiD
        </button>
        <button
          aria-label="Hypebeat"
          className="
                bg-transparent relative z-20 flex w-fit cursor-pointer whitespace-nowrap px-5 py-2
                focus-visible:border-none max-small:mb-0 max-small:shrink-0 max-small:grow-0
                max-small:justify-center 
              "
          onClick={[Function]}
        >
          Hypebeat
        </button>
        <button
          aria-label="Halter"
          className="
                bg-transparent relative z-20 flex w-fit cursor-pointer whitespace-nowrap px-5 py-2
                focus-visible:border-none max-small:mb-0 max-small:shrink-0 max-small:grow-0
                max-small:justify-center 
              "
          onClick={[Function]}
        >
          Halter
        </button>
        <button
          aria-label="BNZ"
          className="
                bg-transparent relative z-20 flex w-fit cursor-pointer whitespace-nowrap px-5 py-2
                focus-visible:border-none max-small:mb-0 max-small:shrink-0 max-small:grow-0
                max-small:justify-center 
              "
          onClick={[Function]}
        >
          BNZ
        </button>
        <button
          aria-label="VUW"
          className="
                bg-transparent relative z-20 flex w-fit cursor-pointer whitespace-nowrap px-5 py-2
                focus-visible:border-none max-small:mb-0 max-small:shrink-0 max-small:grow-0
                max-small:justify-center 
              "
          onClick={[Function]}
        >
          VUW
        </button>
        <div
          className="
            transition-[transform width] ease-[cubic-bezier(0.645, 0.045, 0.355, 1)]
            visible absolute top-0 left-0 z-10 h-[40px] w-[85px] rounded bg-brown delay-100
            duration-200 dark:bg-darkgrey max-small:top-auto max-small:bottom-2.5 max-small:z-20
          "
          style={
            {
              "transform": "translateY(calc(10px)",
            }
          }
        />
        <div
          className="
            visible absolute top-0 left-0 ml-1 h-full
            w-0.5 rounded-sm
            bg-black dark:bg-softwhite max-small:hidden
          "
        />
      </div>
      <div
        className="relative ml-5 w-full max-small:ml-0"
      >
        <div
          className="
                absolute left-0 top-0 m-0 w-full p-0 transition-opacity duration-700 ease-in-out visible relative pt-2.5 opacity-100"
        >
          <div
            aria-label="AplyiD"
          >
            <h2
              className="m-0"
            >
              <span
                className="text-xl font-bold"
              >
                UK Team Lead
              </span>
              <span
                className="text-xl font-bold"
              >
                 @ 
                AplyiD
              </span>
            </h2>
            <p
              className="mt-1 mb-4"
            >
              2022/02 - Present
            </p>
            <div
              className="pb-4"
            >
              APLYiD is a trusted & accredited digital ID solution (AML + KYC) designed to streamline your customer onboarding with a first-class user experience
            </div>
            <ul
              className="m-0 list-none p-0"
            >
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Currently leading all software development initiatives for the UK office
        which involves managing and participating in product/feature development
        from conception and planning to production deployment.
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Building desktop and mobile optimised web applications in <b>Vue.JS</b>
        and <b>Ruby on Rails</b> with an emphasis on the development process to
        produce high quality code. Key aspects of this development process
        involve supporting product managers, performing technical review and
        knowledge sharing.
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Communicated closley with existing and potential clients to provide
        technical support for key API intgerations and with partners to explore
        possible business opportunities.
      </p>",
                  }
                }
              />
            </ul>
          </div>
        </div>
        <div
          className="
                absolute left-0 top-0 m-0 w-full p-0 transition-opacity duration-700 ease-in-out invisible opacity-0"
        >
          <div
            aria-label="AplyiD"
          >
            <h2
              className="m-0"
            >
              <span
                className="text-xl font-bold"
              >
                Software Engineer
              </span>
              <span
                className="text-xl font-bold"
              >
                 @ 
                AplyiD
              </span>
            </h2>
            <p
              className="mt-1 mb-4"
            >
              2020/01 - 2022/02
            </p>
            <div
              className="pb-4"
            >
              APLYiD is a trusted & accredited digital ID solution (AML + KYC) designed to streamline your customer onboarding with a first-class user experience
            </div>
            <ul
              className="m-0 list-none p-0"
            >
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Currently performing full stack development utilising <b>Vue.JS</b>,
        Angular.JS and <b>Ruby on Rails</b> to create KYC/AML and financial
        onboarding products from inception and design to deployment and
        maintenance
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Successfully <b>lead major development initiatives</b> to grow the
        product and company. This resulted in our development team of 2 and
        organisation of 13 to become a strong development team of 7 and an
        organisation of over 50 spanning across 3 timezones.
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Collaborated closley with the founders of the company, key sales staff,
        developers, designers, third party vendors and customer support to
        navigate complex AML regulation and data policies to produce slick
        onboarding products.
      </p>",
                  }
                }
              />
            </ul>
          </div>
        </div>
        <div
          className="
                absolute left-0 top-0 m-0 w-full p-0 transition-opacity duration-700 ease-in-out invisible opacity-0"
        >
          <div
            aria-label="Hypebeat"
          >
            <h2
              className="m-0"
            >
              <span
                className="text-xl font-bold"
              >
                Software Engineer
              </span>
              <span
                className="text-xl font-bold"
              >
                 @ 
                Hypebeat
              </span>
            </h2>
            <p
              className="mt-1 mb-4"
            >
              2019/04 - 2019/09
            </p>
            <div
              className="pb-4"
            >
              A social music application to aid artists in optimising their reach and influence using artifical intelligence
            </div>
            <ul
              className="m-0 list-none p-0"
            >
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Worked closley with the founders of the company in an agile team to
        develop and expand the back-end of the main product using <b>Koa.JS</b>
        which included Javascript, Typescript, HTML and CSS
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "Participated in key strategic meetings regarding business strategy and product design",
                  }
                }
              />
            </ul>
          </div>
        </div>
        <div
          className="
                absolute left-0 top-0 m-0 w-full p-0 transition-opacity duration-700 ease-in-out invisible opacity-0"
        >
          <div
            aria-label="Halter"
          >
            <h2
              className="m-0"
            >
              <span
                className="text-xl font-bold"
              >
                Intern Software Developer
              </span>
              <span
                className="text-xl font-bold"
              >
                 @ 
                Halter
              </span>
            </h2>
            <p
              className="mt-1 mb-4"
            >
              2018/11 - 2019/02
            </p>
            <div
              className="pb-4"
            >
              Halter is a solar-powered GPS enabled smart collar, which guides cows around a farm using sound and vibrations, allowing farmers to automate herd movements and create virtual fences
            </div>
            <ul
              className="m-0 list-none p-0"
            >
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "Worked in two high performing agile teams which developed the minimal viable product of the company under tight product deadlines",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Developed and expanded the back-end and <b>cloud infrastructure</b> of
        the main product using Nest.JS (JavaScript + TypeScript), AWS, Swagger,
        OpenAPI and Docker
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Developed the front-end of the company by creating various internal
        support tools using the <b>React.JS</b> framework
      </p>",
                  }
                }
              />
            </ul>
          </div>
        </div>
        <div
          className="
                absolute left-0 top-0 m-0 w-full p-0 transition-opacity duration-700 ease-in-out invisible opacity-0"
        >
          <div
            aria-label="BNZ"
          >
            <h2
              className="m-0"
            >
              <span
                className="text-xl font-bold"
              >
                Intern Front End Developer
              </span>
              <span
                className="text-xl font-bold"
              >
                 @ 
                BNZ
              </span>
            </h2>
            <p
              className="mt-1 mb-4"
            >
              2017/11 - 2018/02
            </p>
            <div
              className="pb-4"
            >
              Bank of New Zealand is one of New Zealand's big four banks
            </div>
            <ul
              className="m-0 list-none p-0"
            >
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Developed front-end software using the <b>React.JS</b> framework
        alongside technologies such as JavaScript, HTML, CSS, JSS and Node.JS to
        create various visual frontend components
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Worked closely with experienced <b>product designers and owners</b> in a
        small agile team requiring tight communication
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "Partially worked in an agile team to test the banks internal help and support system via automated unit tests and test plans using XMind and Postman",
                  }
                }
              />
            </ul>
          </div>
        </div>
        <div
          className="
                absolute left-0 top-0 m-0 w-full p-0 transition-opacity duration-700 ease-in-out invisible opacity-0"
        >
          <div
            aria-label="VUW"
          >
            <h2
              className="m-0"
            >
              <span
                className="text-xl font-bold"
              >
                Student & Tutor
              </span>
              <span
                className="text-xl font-bold"
              >
                 @ 
                VUW
              </span>
            </h2>
            <p
              className="mt-1 mb-4"
            >
              2016 - 2020
            </p>
            <div
              className="pb-4"
            >
              Victoria University of Wellington is a university in Wellington, New Zealand. The university has a strong software engineering school and an impressive reputation regarding artifical intelligence research
            </div>
            <ul
              className="m-0 list-none p-0"
            >
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "Graduated with a Bachelor of Engineering in Software Engineering with first class honours",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Key study interests included <b>artificial intelligence</b>, software
        ethics and <b>organisational culture</b>
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "<p>
        Completed a year long thesis which involved developing a series of
        <b>IoT</b> connected radios for the purpose of tracking birds within
        dense native bush
      </p>",
                  }
                }
              />
              <li
                className="relative mt-4 pl-8 pb-2.5 before:absolute before:left-0 before:content-['→']"
                dangerouslySetInnerHTML={
                  {
                    "__html": "Worked as a computer science tutor helping students to understand algorithms and data structures",
                  }
                }
              />
            </ul>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
